--- a/linuxdoc/rstFlatTable.py
+++ b/linuxdoc/rstFlatTable.py
@@ -327,7 +327,7 @@ def parseCellItem(self, cellItem):
         cspan = rspan = 0
         if not len(cellItem):  # pylint: disable=len-as-condition
             return cspan, rspan, []
-        for elem in cellItem[0]:
+        for elem in list(cellItem.findall(lambda n: isinstance(n, (colSpan, rowSpan)))):
             if isinstance(elem, colSpan):
                 cspan = elem.get("span")
                 elem.parent.remove(elem)
